hw/mcu/dialog/da1469x/include/mcu/da1469x_snc.h

```diff
@@ -0,0 +1,485 @@
+#define SNC_CMD_INC(addr, inc_by_4)     \
+    (uint32_t)((SNC_OPCODE_INC << 28) + (inc_by_4 << 19) + SNC_ADDR(addr))
+#define SNC_CMD_INC_BY_1(addr)  SNC_CMD_INC(addr, 0)
+#define SNC_CMD_INC_BY_4(addr)  SNC_CMD_INC(addr, 1)
+
+/*
+ * Delay Instruction
+ *
+ * Delay for up to 255 LP clock ticks.
+ *
+ *      -------------------------------
+ *      | opcode (8) |   N/C  | Delay |
+ *      -------------------------------
+ *      |  31 - 28   | 27 - 8 | 7 - 0 |
+ *      -------------------------------
+ */
+#define SNC_OPCODE_DEL      (8)
+#define SNC_CMD_DEL(ticks)  (uint32_t)((SNC_OPCODE_DEL << 28) | (ticks & 0xff))
+
+/*
+ * Sleep instruction
+ *
+ * This instruction is used to halt program execution. Will
+ * generate a signal pulse to PDC and power down the
+ * sensor node controller.
+ *
+ * There are no operands for this instruction.
+ *
+ *      -----------------------
+ *      | opcode (9) |   N/C  |
+ *      -----------------------
+ *      | 31  -  28  | 27 - 0 |
+ *      -----------------------
+ */
+#define SNC_OPCODE_SLP      (9) /* No operands */
+#define SNC_CMD_SLEEP()     (uint32_t)(SNC_OPCODE_SLP << 28)
+
+/*
+ * API NOTES:
+ *
+ * 1) The SNC API are not protected by critical sections. If any of these
+ * API are called by more than one task or inside an ISR they need to be
+ * protected.
+ *
+ * 2) API with _sw_ are intended to be used when the host processor has control
+ * of the SNC (as opposed to the PDC). Typically these API are for debugging as
+ * the PDC usually controls the SNC.
+ */
+
+/**
+ * initialize the SNC for software control
+ *
+ * Called when the host processor wants control of the SNC (PDC
+ * no longer controls SNC).
+ *
+ * @return int 0: success, -1 error.
+ */
+int snc_sw_init(void);
+
+/**
+ * snc sw start
+ *
+ * Starts the SNC. Note that the user should have called
+ * snc_sw_load prior to starting the SNC via software control.
+ *
+ * @return int 0: success -1: error
+ */
+int snc_sw_start(void);
+
+/**
+ * snc sw stop
+ *
+ * Stops the SNC from running a program. This should only be
+ * called when the SNC is under software control.
+ *
+ * @return int 0: success -1: error
+ */
+int snc_sw_stop(void);
+
+/**
+ * Checks if the SNC program has finished
+ *
+ * @return int 0: not finished 1: finished
+ */
+int snc_program_is_done(void);
+
+/**
+ * snc irq config
+ *
+ * Configures the SNC to interrupt the host processor and/or PDC
+ * when SNC generates an interrupt.
+ *
+ * @param mask One or more of the following:
+ *  SNC_IRQ_MASK_NONE: Do not interrupt host or PDC
+ *  SNC_IRQ_MASK_HOST: Interrupt host processor (CM33)
+ *  SNC_IRQ_MASK_PDC: Interrupt PDC
+ *
+ * @return int 0: success -1: invalid mask parameter
+ *
+ * NOTES:
+ *
+ * 1) The IRQ configuration cannot be changed if there is a
+ * pending IRQ. The IRQ must be cleared in that case. This
+ * function will automatically clear the IRQ if that is the case
+ * and will not report an error.
+ */
+int snc_irq_config(uint8_t mask);
```

Comment:
I think I'd like to have this code to also install IRQ handler (if mask is is set to interrupt the host), have that IRQ handler clear the SNC as interrupt source, and call a user-registered callback.
E.g this snc_irq_config() could take pointer to that callback routine as it's argument.